Tidy up CHANGES.rst for 1.2

astrofrog opened the pull request:
I'm currently going through and making sure the changelog is consistent with labels and milestones for merged PRs, so this is a work in progress. As part of this, I've created a new label called ``no-changelog-entry-required`` which we should add to PRs that deliberately don't have a changelog entry (including this PR). This will make it easier to check for consistency in an automated way (which is what I'm doing).

cc @eteq (just so we don't duplicate effort)

Things to raise on astropy-dev:
* [x] use the new no-changelog-entry-needed label
* [x] always give number of PR (in addition to issues if relevant)

MSeifert04 commented on CHANGES.rst:
```diff
@@ -84,7 +84,8 @@ New Features
   - Added ``NDDataRef`` that implements ``NDData`` together with all
     currently avaiable mixins. This class does not implement additional
-    attributes, methods or a numpy.ndarray-like interface like ``NDDataArray``.
+    attributes, methods or a numpy.ndarray-like interface like ``NDDataArray``. 
```
Found a trailing whitspace in this line. And thank you for including the PR number. :+1:

astrofrog commented:
We are re-milestoning things that were 1.1.3 to 1.2.0 since we won't do a 1.1.3 release, but for the record here are the PRs that were originally milestoned as 1.1.3 and hadn't been backported yet:

```
[#4930][36ed826a98c3dc54e1d7fc223e87968f80bbd026]: Make Table inherit primary_key attribute from its parent Table.
[#4910][be988de68fe94b0667fb889595e21722e27b9564]: Ascii sextractor parse
[#4808][8f0b28f68aea347dd01cf991aa212f762b5c846e]: Fix sigma_clipped_stats when no mask is used.
[#4792][5337729eaf9e6ddf70de77601311ed6410d237fc]: Windows msvc14 has no useful find_msvcrt() method anymore.
[#4782][455f0fa83648ca4c71f547c39a45c9526ceea88c]: Bugfix for issue 4781
[#4777][4fe85d5d39dbbe7e3365abce709d16e9585d1f97]: Fixed indentation problem with author name
[#4768][8fe60215b22b758a149e302401754b81c14b924e]: Update references.txt
[#4766][9b33644a4ac8ea26ab9594f0e73caaf2d80e129a]: Update equivalencies.py
[#4742][5b8a0ee1b2ed4af9eaff1a5cfc8805314ff89df5]: Set is_image = False for GroupsHDU
[#4726][1f72cec3bb1f35cc15be7b2f67ae63e893b694b1]:  Fix sigma_clipped_stats to use the axis argument
[#4722][0133ab0f5ff60ecc7f1ebd7c9d83428d0e9c9653]: Ensure we don't use floats for number of elements in linspace.
[#4720][249b4815f3441c2f1e7a24945bf543f56609a2d8]: Fix AiryDisk2D, Sersic1D, and Sersic2D models so they can be used in compound models
[#4683][b7989c7d9b40c066f4f7fbd3753ec2ca5e2aeec4]: fixed wrong rendered changelog entry and contributor name [skip ci]
```

eteq commented:
Looks good!  Merging this now to make life easier on newer things that may need to modify the changelog.
